resolver/src/p2p/mod.rs

```diff
@@ -0,0 +1,47 @@
+//! Makes and responds to requests using the P2P network.
+
+use std::future::Future;
+
+use commonware_cryptography::Array;
+use futures::channel::oneshot;
+
+#[cfg(test)]
+pub mod mocks;
+
+pub mod peer;
+mod wire {
+    include!(concat!(env!("OUT_DIR"), "/wire.rs"));
+}
+
+/// Type of data resolved by the p2p network.
+/// This is a blob of bytes that is opaque to the resolver.
+pub type Value = Vec<u8>;
+
+/// The interface responsible for serving data requested by the network.
+pub trait Producer: Clone + Send + 'static {
+    /// Type used to uniquely identify data.
+    type Key: Array;
+
+    /// Type of data returned by the producer.
+    type Value;
+
+    /// Serve a request received from the network.
+    fn produce(
+        &mut self,
+        key: Self::Key,
+    ) -> impl Future<Output = oneshot::Receiver<Self::Value>> + Send;
+}
+
+/// The interface responsible for managing the list of peers that can be used to fetch data.
+pub trait Director: Clone + Send + 'static {
+    /// Type used to uniquely identify peers.
+    type PublicKey: Array;
+
+    /// Returns the current list of peers that can be used to fetch data.
+    ///
+    /// This is also used to filter requests from peers.
+    fn peers(&self) -> Vec<Self::PublicKey>;
```

Comment:
Recommend doing `&Vec<Self::PublicKey>` here so we can reuse the same vec: https://github.com/commonwarexyz/monorepo/blob/1656a0456e3770a2f48eb7ec6b91cf36ae4893eb/consensus/src/lib.rs#L131

Reply:
I may have missed this in broadcast, so double-check we do it there too.